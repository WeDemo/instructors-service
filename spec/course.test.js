import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Course from '../client/src/components/course.jsx';

const renderer = require('react-test-renderer');
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('Course Component Snapshot', () => {
  const defaultProps = {
    cInfo: {
      photo_url: 'google.com',
      lectures: 10,
      num_hours: 10,
      rating: 3.0,
      reviews: 100,
      full_price: 500.00,
      disc_price: 50.00,
      course_name: 'Testing React'
    },
    instInfo: {
      inst_name: "Charlie",
      photo_url: 'google.com'
    }
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(< Course {...defaultProps} />);
  });

  it('renders Course component', () => {  
    const CourseTest = (props) => <Course {...defaultProps} {...props} />;
    const CourseTestComponent = renderer.create(<CourseTest />).toJSON();
    expect(CourseTestComponent).toMatchSnapshot();
  });

  it('should render a div', () => {  
    expect(wrapper.find('div')).toBeDefined();     
  });
  
  it('should render an instructor_course class', () => {
    expect(wrapper.exists('.instructorCourse')).toEqual(true);
  });

  it('should render two hearts', () => {
    expect(wrapper.find('.fa-heart').length).toEqual(2);
  });

  it('should change state on mouse enter and mouse leave', () => {
    expect(wrapper.state('hover')).toBe(false);
    wrapper.simulate("mouseEnter");
    expect(wrapper.state('hover')).toBe(true);
    wrapper.simulate("mouseLeave");
    expect(wrapper.state('hover')).toBe(false);
  });
});