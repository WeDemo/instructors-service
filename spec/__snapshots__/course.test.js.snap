// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Component Snapshot renders Course component 1`] = `
<div
  className="instructor_course"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="course_photo_section"
    style={
      Object {
        "backgroundImage": "url(google.com)",
      }
    }
  >
    <i
      className="far fa-heart fa-lg big-heart unclicked"
    />
    <i
      className="fas fa-heart small-heart unclicked"
      onClick={[Function]}
    />
  </div>
  <div
    className="course_text_section"
  >
    <span
      className="course_name"
    >
      Testing React
    </span>
    <div
      className="course_instructor"
    >
      Charlie
    </div>
    <div
      className="course_stats_bar"
    >
      <img
        className="empty-stars"
        src="https://s3.us-east-2.amazonaws.com/udemy-demo-tarik/emptyStars.png"
      />
      <div
        className="stars-container"
        style={
          Object {
            "width": "45px",
          }
        }
      >
        <img
          className="full-stars"
          src="https://s3.us-east-2.amazonaws.com/udemy-demo-tarik/filledStars.png"
        />
      </div>
      <span
        className="course_rating"
      >
        3
      </span>
      <span
        className="course_reviews"
      >
        (
        100
        )
      </span>
    </div>
    <div
      className="course_price_bar"
    >
      <span
        className="course_disc_price"
      >
        $
        50
      </span>
      <span
        className="course_full_price"
      >
        $
        500
      </span>
    </div>
  </div>
</div>
`;
